modules/src/timestamp.rs

```diff
@@ -88,42 +90,48 @@ impl Timestamp {
     /// than the current or if either of the `Timestamp`s is not set.
     pub fn duration_since(&self, other: &Timestamp) -> Option<Duration> {
         match (self.time, other.time) {
-            (Some(time1), Some(time2)) => time1.signed_duration_since(time2).to_std().ok(),
+            (Some(time1), Some(time2)) => time1.duration_since(time2).ok(),
             _ => None,
         }
     }
 
-    /// Convert a `Timestamp` from [`chrono::DateTime<Utc>`].
-    pub fn from_datetime(time: DateTime<Utc>) -> Timestamp {
-        Timestamp { time: Some(time) }
+    /// Convert a `Timestamp` to `u64` value in nanoseconds. If no timestamp
+    /// is set, the result is 0.
+    ///
+    #[deprecated(since = "0.9.1", note = "use `nanoseconds` instead")]
+    pub fn as_nanoseconds(&self) -> u64 {
+        (*self).nanoseconds()
     }
 
     /// Convert a `Timestamp` to `u64` value in nanoseconds. If no timestamp
     /// is set, the result is 0.
     /// ```
     /// use ibc::timestamp::Timestamp;
+    ///
     /// let max = u64::MAX;
     /// let tx = Timestamp::from_nanoseconds(max).unwrap();
-    /// let utx = tx.as_nanoseconds();
+    /// let utx = tx.nanoseconds();
     /// assert_eq!(utx, max);
     /// let min = u64::MIN;
     /// let ti = Timestamp::from_nanoseconds(min).unwrap();
-    /// let uti = ti.as_nanoseconds();
+    /// let uti = ti.nanoseconds();
     /// assert_eq!(uti, min);
+    /// let tz = Timestamp::default();
+    /// let utz = tz.nanoseconds();
+    /// assert_eq!(utz, 0);
     /// ```
-    pub fn as_nanoseconds(&self) -> u64 {
+    pub fn nanoseconds(self) -> u64 {
         self.time.map_or(0, |time| {
-            let s = time.timestamp();
+            let t: OffsetDateTime = time.into();
+            let s = t.unix_timestamp_nanos();
             assert!(s >= 0, "time {:?} has negative `.timestamp()`", time);
-            let s: u64 = s.try_into().unwrap();
-
-            util::assemble_in_nanos(s, time.timestamp_subsec_nanos())
+            s.try_into().unwrap()
         })
     }
 
-    /// Convert a `Timestamp` to an optional [`chrono::DateTime<Utc>`]
-    pub fn as_datetime(&self) -> Option<DateTime<Utc>> {
-        self.time
+    /// Convert a `Timestamp` to an optional [`OffsetDateTime`]
+    pub fn into_datetime(self) -> Option<OffsetDateTime> {
```

Comment:
I wonder if we'd be more interested in getting `tendermint::Time` here?
That would also get rid of a public dependency on `time`.

Reply:
Outside of ics07, we are not supposed to depend on Tendermint-specific code but I guess we can make an exception for the `Time` struct since it's general purpose?

Reply:
In my mad scientist's secret world assimilation plan, we'll migrate to a common domain types crate that will provide idiomatic Rust support for Google's well-known protobuf types, and `tendermint::Time` will be morphed into one of the types from that yet unpublished crate. Until then, it could be whatever seems most practical at the moment.